Tenancy support

ASACookCodes opened the pull request:
Current state of Burp integration:
1. Login and config data on different tabs in Mindgard extension UI
2. Setting the config to correct values (sandbox) confirmed working: token issued

Known issues:
1. Setting the config to incorrect values (example.com) fails silently. ‼️ Bad UX
2. Project ID validation does not seem to work as expected. It should be blowing up an error. ‼️ Bad UX

Future work:
1. Test config should be inaccessible unless logged in
2. Token.txt should be voided if login server info changes - no longer valid

ASACookCodes committed:
First cut - custom tenancy fields.

- New settings record without tenancy details defaults to Constants.
- New UI elements to support custom tenancy details
- Sandbox connection factory refactored and documented, with api endpoint pulled in from MindgardSettings as opposed to a constant.
- Other javadocs

ASACookCodes committed:
- UI has login button and test configuration tabs
- Login flow removed from the initialization of the extension
- Removed MindgardSettingsManager as not needed
- Settings passed in to MindgardAuthentication, because the api endpoint for tests is pulled from the url field on the record

ASACookCodes committed:
UI refactoring

Laukei commented on src/main/java/ai/mindgard/sandbox/wsapi/SandboxConnectionFactory.java:
```diff
@@ -11,72 +11,99 @@
 import java.util.stream.Collectors;
 
 import static ai.mindgard.JSON.fromJson;
 import static ai.mindgard.JSON.json;
 
+/**
+ * Factory class responsible for creating and managing connections to the Mindgard Sandbox WebSocket API.
+ * <p>
+ * This class provides methods to initialize and validate connections, as well as to handle authentication and
+ * configuration for the Mindgard sandbox environment. It supports both default and parameterized construction,
+ * allowing for flexible dependency injection of HTTP clients, body publishers, and WebSocket client factories.
+ * </p>
+ *
+ * <p>
+ * Usage example:
+ * <pre>
+ *     SandboxConnectionFactory factory = new SandboxConnectionFactory();
+ *     SandboxConnection connection = factory.connect(mindgard, auth, settings, logger);
+ * </pre>
+ * </p>
```
HTML tags in here - is that normal?

Laukei commented on src/main/java/ai/mindgard/sandbox/wsapi/SandboxConnectionFactory.java:
```diff
@@ -139,4 +167,30 @@ public boolean validateProject(String projectID) {
+     * @param includedOrExcluded The comma-separated string
+     * @return List of trimmed strings, or null if input is null or empty
+     */
+    private List<String> commaSeperatedToList(String includedOrExcluded) {
+        if (includedOrExcluded == null || includedOrExcluded.isEmpty()) {
+            return null;
+        }
+        return Arrays.stream(includedOrExcluded.split(","))
+                .map(String::trim)
+                .collect(Collectors.toList());
+    }
```
Feels very "util"-y - could this live somewhere more generic?

Laukei commented on src/main/java/ai/mindgard/LoginTab.java:
```diff
@@ -0,0 +1,102 @@
+        loginButtonPanel = new JPanel();
+        JButton loginButton = new JButton("Login");
+        loginButton.addActionListener((actionEvent) -> {
+            var auth = new MindgardAuthentication(settings);
```
I think we want to enforce saving the settings before using them

Laukei commented on src/main/java/ai/mindgard/MindgardSettingsManager.java:
```diff
@@ -1,18 +0,0 @@
- * Mindgard Settings Manager
- * Used to synchronise both the MindgardSettingsUI and MindgardSettings views of the state of the settings.
- */
-public class MindgardSettingsManager {
```
`currentSettings` private internal state; change setSettings/getSettings to set/get from this. Unsure whether this can be static or whether we intantiate the manager and pass that around.

ASACookCodes committed:
MindgardSettingsManager re-introduced and used in appropriate places.

Co-authored-by: Rob Heath <robert.heath@mindgard.ai>

Laukei commented:
Parked for other more important work

Laukei committed:
Fix validation of project ID

Modify the validation logic to throw an error, and catch it in the
settings.

@Co-authored-by: Andrew Cook <andrew.cook@mindgard.ai>

Laukei committed:
Use MindgardToken to store the auth token

The existing method of storing tokens is to write to token.txt in the
MindgardAuthentication model using the MindgardSettings file getter.
This change moves the token into its own class, controlled by the
MindgardSettingsManager, which allows simpler reading/writing and
persistence of state wherever the manager is loaded.

Laukei committed:
Modify MindgardAuthenticationTest not to write to disk

This was overwriting the settings for Burp every time the tests were
run.

Laukei committed:
Add validLogin boolean method to settings manager

This will allow us to control flow through Mindgard's UI by determining
whether or not a token is present, and if so, whether it is for the
currently defined Mindgard base URL.

I have also removed the redundant functionality of passing a filename
into the loadOrCreate method.

Laukei committed:
Add visual indicator of login status

This adds a text indicator of the login status for a given tenant. It
also flashes a warning if you try to use the generator without being
logged in.

Laukei committed:
Quality of life improvements for logging in: timeouts, warnings

This modifies the login behaviour in a few key ways:
1. It adds a one minute timeout for a user authing. If a user does not
   auth before that time, a warning is shown and the process resets.
   This prevents Burp from hanging indefinitely if the customer
   rejects the key request.
2. It adds a warning if there are changes to settings that have not been
   saved when a login is attempted. This prevents a user from changing
   the server URL and then attempting to sign in with the server address
   incorrectly set.

Laukei committed:
Add negative space above settings panel objects

This makes the settings panel look more sensible.
